python/lsst/sims/photUtils/Sed.py

```diff
@@ -517,36 +522,32 @@ def readSED_flambda(self, filename, name=None):
             sourceflambda = numpy.copy(cached_source[1])
 
         if cached_source is None:
+            # Read source SED from file - lambda, flambda should be first two columns in the file.
+            # lambda should be in nm and flambda should be in ergs/cm2/s/nm
+            dtype = numpy.dtype([('wavelen', float), ('flambda', float)])
             try:
-                f = gzip.open(gzipped_filename, 'r')
+                data = numpy.genfromtxt(gzipped_filename, dtype=dtype)
             except IOError:
                 try:
-                    f = open(unzipped_filename, 'r')
-                except Exception as e:
-                    # append our message to the message of the error that was actually raised
-                    # code taken form
-                    # http://stackoverflow.com/questions/6062576/adding-information-to-an-exception
-                    new_exception = type(e)(str(e) +
-                                            "\n\nError reading sed file %s; "
-                                            "it may not exist, "
-                                            "or be improperly formatted "
-                                            "(if the file name ends in .gz it should be gzipped; "
-                                            "if not, it should just be a text file)" % filename)
-                    raise type(e), new_exception, sys.exc_info()[2]
+                    data = numpy.genfromtxt(unzipped_filename, dtype=dtype)
+                except Exception as err:
+                    # see
+                    # http://stackoverflow.com/questions/
+                    # 9157210/how-do-i-raise-the-same-exception-with-a-custom-message-in-python
+                    new_args = [err.args[0] + \
+                                "\n\nError reading sed file %s; " % filename \
+                                + "it may not exist, " \
+                                + "or be improperly formatted " \
+                                + "(if the file name ends in .gz it should be gzipped; " \
+                                + "if not, it should just be a text file)"]
```

Comment:
Nitpick: I'm not sure I quite understand the problem/message matchup here. 
There was a try/except to read from either the gzipped or unzipped filename, but apart from the filename, the command is the same. So it's kind of irrelevant whether if the file ends in .gz it's actually zipped or not isn't it?